specifyweb/frontend/js_src/lib/components/Interactions/PrepDialog.tsx

```diff
@@ -192,6 +199,10 @@ export function PrepDialog({
               'disposalPreparations',
               items as RA<SpecifyResource<DisposalPreparation>>
             );
+            toTable(interaction, 'ExchangeOut')?.set(
+              'exchangeOutPreps',
+              items as RA<SpecifyResource<ExchangeOutPrep>>
+            );
```

Comment:
The following lines can be approached dynamically, rather than checking for each individual interaction: 
https://github.com/specify/specify7/blob/a2136ba121d518d4ae7db35b22b2dd6fc26aaa87/specifyweb/frontend/js_src/lib/components/Interactions/PrepDialog.tsx#L187-L205 

But first, we need to find a way to specify a type of all the base Interaction tables which have associated TableNamePreparation relationships. 

We would preferably find some way to extract the value directly from the `Tables` type, but I was unable to find a clean solution. 

Instead, I use the type `AnyInteractionPreparation` to get the possible table names of the InteractionPreparations table and remove the `Prep*` suffix, extracting the prefix without the preparation part. 

```ts
type ExtractInteraction<T extends string> =
  T extends `${infer Prefix}Prep${string}` ? Prefix : never;

export type InteractionsWithPreps = Tables[ExtractInteraction<
  AnyInteractionPreparation['tableName']
>];
```

Now, any place which needs the Interaction part of an AnyInteractionPreparation, we can replace with the `InteractionsWithPreps` preps. 

---

Instead of the `toTable` check for each interaction and setting the hardcoded field name to the value of the items, a helper function can be defined to handle the general case: 

```ts
function setPreparationItems(
  interaction: SpecifyResource<InteractionsWithPreps>,
  items: RA<SpecifyResource<AnyInteractionPreparation>>
): void {
  const preparationField = interaction.specifyTable.relationships.find(
    (relationship) =>
      interactionPrepTables.includes(relationship.relatedTable.name)
  );
  interaction.set(preparationField?.name, items);
}
```

Now all of the `toTable(...).set(somePrep, items)` can be removed and replaced with a single `setPreparationItems(interaction, items)`